src/app/parcels/ActionBar/ActionModals/SlotChangeModal.tsx

```diff
@@ -0,0 +1,153 @@
+    PackingSlotsLabelsAndValues,
+    fetchPackingSlotsInfo,
+} from "@/common/fetch";
+import DropdownListInput from "@/components/DataInput/DropdownListInput";
+import { UpdateParcelError } from "../../form/submitFormHelpers";
+import { getUpdateErrorMessage, packingDateOrSlotUpdate } from "./CommonDateAndSlot";
+
+interface SlotInputProps {
+    packingSlotsLabelsAndValues: PackingSlotsLabelsAndValues;
+    setSlot: (slot: string) => void;
+}
+
+const SlotChangeInput: React.FC<SlotInputProps> = ({ packingSlotsLabelsAndValues, setSlot }) => {
+    return (
+        <>
+            <Heading>What date would you like to change to?</Heading>
+            <DropdownListInput
+                selectLabelId="packing-slot-select-label"
+                labelsAndValues={packingSlotsLabelsAndValues}
+                listTitle="Packing Slot"
+                defaultValue=""
+                onChange={(event) => {
+                    setSlot(event.target.value);
+                }}
+            />
+        </>
+    );
+};
+
+const SlotChangeModal: React.FC<ActionModalProps> = (props) => {
+    const [actionCompleted, setActionCompleted] = useState(false);
+    const [errorMessage, setErrorMessage] = useState<string | null>(null);
+    const [successMessage, setSuccessMessage] = useState<string | null>(null);
+
+    const [packingSlots, setPackingSlots] = useState<[string, string][]>([]);
+    const [slot, setSlot] = useState<string>("");
+    const [displayModal, setDisplayModal] = useState<boolean>(true);
+    const [warningMessage, setWarningMessage] = useState<string>("");
+
+    (async () => {
+        const { data: packingSlotsData, error: packingSlotsError } =
+            await fetchPackingSlotsInfo(supabase);
+        if (packingSlotsError) {
+            setErrorMessage(
+                `Failed to fetch packing slots data. Log Id: ${packingSlotsError.logId}`
+            );
+            setDisplayModal(false);
+            return;
+        }
+        setPackingSlots(packingSlotsData);
+    })();
+
+    const onSlotChange = async (): Promise<void> => {
+        if (slot === "") {
+            setWarningMessage("Please choose a valid packing slot.");
+            return;
+        }
+        let packingSlotUpdateError: FetchParcelError | UpdateParcelError | null = null;
+        for (const parcel of props.selectedParcels) {
+            (async () => {
+                const { error } = await packingDateOrSlotUpdate("packingSlot", slot, parcel);
+                if (error) {
+                    packingSlotUpdateError = error;
+                    return;
+                }
+            })();
+        }
+
+        let newPackingSlotText: string = "";
+        packingSlots.forEach((packingSlot) => {
+            if (packingSlot[1] === slot) {
+                newPackingSlotText = packingSlot[0];
+            }
+        }); //since 'slot' is the key for the packing slot, this algorithm is used to just recover the value, i.e. 'AM', 'PM, etc.
```

Comment:
packingSlots is a list of all available slots with the structure `{{"AM", <AM key>}, {"PM", <PM key>}, ...}`. These change based on the allowed packing slots as set by the admin. Strangely enough the only information that is retained when retrieving the user input from the dropdown is the key, which is stored in `slot`. This algorithm is just a quick and dirty way to get the corresponding string value back. 

The code where the information on the string (i.e. "AM") is lost is in components > DataInput > DropdownListInput.tsx where in the MenuItem component `label` is not stored. It might be possible to pass both `label` and `value` into the component (I will have to check) - this would require a change on everything that uses dropdown lists. Everything else seems to be compatible which is why I just used the above algorithm, but let me know if you want me to find a solution that is more robust :)